pkg/controller/jaeger/upgrade.go

```diff
@@ -0,0 +1,25 @@
+package jaeger
+
+import (
+	v1 "github.com/jaegertracing/jaeger-operator/pkg/apis/jaegertracing/v1"
+	"github.com/jaegertracing/jaeger-operator/pkg/upgrade"
+	"github.com/jaegertracing/jaeger-operator/pkg/version"
+)
+
+func (r *ReconcileJaeger) applyUpgrades(jaeger v1.Jaeger) error {
+	currentVersions := version.Get()
+
+	if len(jaeger.Status.Version) > 0 {
```

Comment:
Who sets/updates the version in the status object?

Reply:
The Operator sets whenever a new instance is created. This way, once the Operator is updated, it knows which instances needs to be brought up to the current/new version.